### PR TITLE
[3.4.4] Fix data convertion for BigInteger values that stored in DB as String

### PR DESCRIPTION
**Pull request by ShvaykaD**
## Pull Request description

issue: #8001

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.
  
## Front-End feature checklist

- [x] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [x] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [x] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

File: dao/src/test/java/org/thingsboard/server/dao/sql/query/EntityDataAdapterTest.java [hunk 1]
```diff
@@ -25,6 +25,7 @@ public class EntityDataAdapterTest {
     public void testConvertValue() {
         assertThat(EntityDataAdapter.convertValue("500")).isEqualTo("500");
         assertThat(EntityDataAdapter.convertValue("500D")).isEqualTo("500D"); //do not convert to Double !!!
-        assertThat(EntityDataAdapter.convertValue("0101010521130565")).isEqualTo("0101010521130565");
+        assertThat(EntityDataAdapter.convertValue("0101010521130565")).isEqualTo("0101010521130565"); //do not convert to Double !!!
```
Comment (ashvayka) [inline dao/src/test/java/org/thingsboard/server/dao/sql/query/EntityDataAdapterTest.java:28]: You have added check for "Simple double" that contained ".", but I don't see new tests values that contain ".". When do you return Double as double?

Comment (ShvaykaD) [reply dao/src/test/java/org/thingsboard/server/dao/sql/query/EntityDataAdapterTest.java:28]: `EntityDataAdapter.convertValue` value method always accepts strings as input parameter. 

Simple double added to verify that BigInteger value like `89010303310033979663` will not be converted to: 8.901030331003398E19. 

Please check the example added in the github issue: 

https://github.com/thingsboard/thingsboard/issues/8001

File: dao/src/main/java/org/thingsboard/server/dao/sql/query/EntityDataAdapter.java [hunk 2]
```diff
@@ -92,8 +92,9 @@ static String convertValue(Object value) {
                 }
                 try {
                     double dblVal = Double.parseDouble(strVal);
-                    if (!Double.isInfinite(dblVal)) {
-                        return Double.toString(dblVal);
+                    String doubleAsString = Double.toString(dblVal);
```
Comment (smatvienko-tb) [inline dao/src/main/java/org/thingsboard/server/dao/sql/query/EntityDataAdapter.java:95]: Hey! @ShvaykaD 

Look at a more optimal way to identify a double type (have a `.` char).
Another improvement is to verify the string length to asses for long or double string content (let's say 64 chars)
I hope you can add more use cases for conversion as well.
It is essential code that affects data representation.
image

File: dao/src/test/java/org/thingsboard/server/dao/sql/query/EntityDataAdapterTest.java (hunk 1 above)
Comment (smatvienko-tb) [reply dao/src/test/java/org/thingsboard/server/dao/sql/query/EntityDataAdapterTest.java:28]: @ShvaykaD, how about adding the exact test case for the issued value `89010303310033979663` ?

**Pull request merged**